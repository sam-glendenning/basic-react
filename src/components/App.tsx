import React, { ReactElement } from "react";

/**
 * 
 * @param {string} name - the name to display
 * @returns {string} - the hello message
 */
export const label = (name : string) : string => {
	return `Hello, ${name}!`;
};

function App() : ReactElement {
	return (
		<div className="App">
			<h1>{label("Sam")}</h1>
			<p id="welcome">Welcome to my React app.</p>
		</div>
	);
}

export default App;
