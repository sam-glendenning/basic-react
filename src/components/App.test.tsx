import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import App, { label } from "./App";
import { describe, it } from "@jest/globals";

describe("App", () => {
	it("renders correct text in a header", () => {
		const wrapper : ShallowWrapper = shallow(<App />);
		expect(wrapper.find("h1").text()).toBe("Hello, Sam!");
	});

	it("renders the first paragraph", () => {
		const wrapper : ShallowWrapper = shallow(<App />);
		expect(wrapper.find("p").text()).toBe("Welcome to my React app.");
	});

	it("generates a label", () => {
		const name  = "Dave";
		const l : string = label(name);
		expect(l).toBe("Hello, " + name + "!");
	});
});
