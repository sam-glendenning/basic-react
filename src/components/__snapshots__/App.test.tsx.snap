// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="App"
>
  <h1>
    Hello, Sam!
  </h1>
  <p
    id="welcome"
  >
    Welcome to my React app.
  </p>
</div>
`;
